configure.ac

```diff
@@ -1374,8 +1379,21 @@ fi
 
 dnl Enable -fsanitize=memory late, because interceptors may break linking detection.
 if test "$PHP_MEMORY_SANITIZER" = "yes"; then
-  CFLAGS="$CFLAGS -fsanitize=memory -fsanitize-memory-track-origins"
-  CXXFLAGS="$CXXFLAGS -fsanitize=memory -fsanitize-memory-track-origins"
+  AX_CHECK_COMPILE_FLAG([-fsanitize=memory -fsanitize-memory-track-origins], [
+    CFLAGS="$CFLAGS -fsanitize=memory -fsanitize-memory-track-origins"
+    CXXFLAGS="$CXXFLAGS -fsanitize=memory -fsanitize-memory-track-origins"
+  ], [AC_MSG_ERROR([MemorySanitizer is not available])])
+fi
+
+if test "$PHP_ADDRESS_SANITIZER" = "yes"; then
+  AX_CHECK_COMPILE_FLAG([-fsanitize=undefined,address], [
+    CFLAGS="$CFLAGS -fsanitize=undefined,address -DZEND_TRACK_ARENA_ALLOC"
+    CXXFLAGS="$CXXFLAGS -fsanitize=undefined,address -DZEND_TRACK_ARENA_ALLOC"
+    LDFLAGS="$LDFLAGS -fsanitize=undefined,address"
+    AX_CHECK_COMPILE_FLAG([-fno-omit-frame-pointer], [
+      CFLAGS="$CFLAGS -fno-omit-frame-pointer"
```

Comment:
Don't think it's necessary to check the flag, any unix compiler should support this option...